### DISCUSSION TITLE
Use fzf.fish for tab completion?

### DISCUSSION THREAD

**Discussion by cr-mitmit** [PatrickF1/fzf.fish#217] [Archive]
Hi.

I wanted to get your thoughts on using fzf as a generic completer function?
Should be quite easy to bind \ci (tab) to a custom function that runs fish's `complete` function and pipes the result through fzf. This way you get a nice ability to filter any context aware completion already installed through fzf.

Any thoughts whether or not this is a good idea?

**Comment by mandarvaze** [comment]
I just started using `fzf` recently. So I'm a n00b. `fzf` documentation mentions using `**` to invoke `fzf` anywhere.
@cr-mitmit Is that what you mean by generic completer ?

If yes, that would be awesome. (Unless it already exists, and I just don't know)

**Reply by cr-mitmit** [reply]
Sorta. 
What I really meant is to see if we can use fzf for every use of tab. Binding it only when the token is ** is an interesting thing to test as well.

The use case you described about using ** only works on zsh as far as I know

**Reply by mandarvaze** [reply]
Yeah. I understand `**` is shell specific. (I assumed bash, but may be zsh)

I don't mean necessarily `**` with fish. I meant similar - some keybinding to invoke `fzf` anywhere. It could be tab as well.
`**` was just an example that comes out of the box (but does not work with fish)

**Comment by PatrickF1** [comment]
Hi again, we have a strong candidate PR for tab completion here. Check it out!
https://github.com/PatrickF1/fzf.fish/pull/293
Please upvote this answer or select it as the answer so more people can see it, thanks!

**Comment by PatrickF1** [comment]
After giving a serious try to implement this, I decide to not ever do this. See https://github.com/PatrickF1/fzf.fish/pull/293#issuecomment-1793652344.
